app/scripts/proactive/view/dom.js

```diff
@@ -288,7 +290,26 @@ define(
 
         $("#set-templates-secondary-bucket-select-button").click(function () {
             var bucketName = ($(($("#catalog-set-templates-secondary-bucket-table .catalog-selected-row"))[0])).text();
-            require('StudioApp').views.palleteView.setSecondaryTemplatesBucket(bucketName, false);
+            // Leave modal open if bucket could not be added (normally because it's already in the Palette)
+            if (require('StudioApp').views.paletteView.addPaletteBucketMenu(bucketName, false)){
+                $('#set-templates-secondary-bucket-modal').modal('hide');
+            }
+        });
+
+        // Set default preset on startup
+        if (!localStorage['palettePreset']){
```

Comment:
I don't see where you use localStorage['palettePreset']. What happens if buckets that are saved in localStorage['palettePreset'] (or config.palette_presets) are removed?

Reply:
they are used in `PalettePreview.js`